Sources/MapboxNavigation/SpriteRepository.swift

```diff
@@ -5,13 +5,13 @@ import MapboxDirections
 
 class SpriteRepository {
     // Caching the Sprite and single shield icon images.
-    let spriteCache = ImageCache()
+    var spriteCache = ImageCache()
     // Caching the single legacy shield icon images.
-    let legacyCache = ImageCache()
+    var legacyCache = ImageCache()
     // Caching the metadata info for Sprite.
-    let infoCache =  SpriteInfoCache()
+    var infoCache =  SpriteInfoCache()
     var styleURI: StyleURI = .navigationDay
-    var baseURL: URL = URL(string: "https://api.mapbox.com/styles/v1")!
+    let baseURL: URL = URL(string: "https://api.mapbox.com/styles/v1")!
```

Comment:
Because the `baseURL` won't change right now, it would be better to keep it as a constant inside.